### DISCUSSION TITLE
Stable smartlog output

### DISCUSSION THREAD

**Discussion by justhecuke** [arxanas/git-branchless#1216] [Ideas]
From run-to-run, the order of output of smartlog can change. This makes it difficult to compare smartlogs or use it in automation.

We've written a wrapper around smartlog output to add information about each branch and it is very frustrating that the same information is displayed in different orders on every run.

**Comment by arxanas** [comment]
It is intended to be stable —

https://github.com/arxanas/git-branchless/blob/2923924dfbff07faa61d50771b6604cb18e64bc8/git-branchless-smartlog/src/lib.rs#L309-L311

So if it's not stable for you then it's a bug.

**Reply by justhecuke** [reply]
It isn't, so I guess it's a bug.

**Reply by claytonrcarter** [reply]
I've never found the output to be unstable. Can you share some examples of the output that you're seeing? Also, is it unstable when you run `smartlog` "back to back", or is it possible that you're running other commands in between, commands that may be affecting the repo in some way?

**Reply by justhecuke** [reply]
back-to-back smartlog

```
$ git sl 'branches()'
⋮
◆ 0000001 1d (ᐅ main) msg
┣━┓
┃ ◯ 0000002 1d (a) msg
┣━┓
┃ ◯ 0000003 1d (b) msg
┃ ┃
┃ ◯ 0000004 1d (c) msg
┃ ┃
┃ ◯ 0000005 1d (d) msg
┣━┓
┃ ◯ 0000006 1d (e) msg
┣━┓
┃ ◯ 0000007 19h (f) msg
┣━┓
┃ ⊘ 4 omitted commits
┃ ⋮
┃ ◯ 0000008 1d (g) g is before h in the first run, but after in the second one
⋮
⊘ 1 omitted commit
⋮
◯ 0000009 1d (h) msg


$ git sl 'branches()'
⋮
◆ 9cc2d85 1d (ᐅ main) msg
┣━┓
┃ ◯ 0000002 1d (a) msg
┣━┓
┃ ◯ 0000003 1d (b) msg
┃ ┃
┃ ◯ 0000004 1d (c) msg
┃ ┃
┃ ◯ 0000005 1d (d) msg
┣━┓
┃ ◯ 0000006 1d (e) msg
┣━┓
┃ ◯ 0000007 19h (f) msg
┣━┓
┃ ⊘ 1 omitted commit
┃ ⋮
┃ ◯ 0000009 1d (h) msg
⋮
⊘ 4 omitted commits
⋮
◯ 0000008 1d (g) g is before h in the first run, but after in the second one
```

**Reply by arxanas** [reply]
You can file a bug for this if you want to track it, then. I haven't been able to reproduce locally with `git branchless smartlog 'branches()'`, despite having sibling branches in my repo.
